Toggling empty dirs raises error

In projects bigger than `config.max_project_files`, toggling empty directories in the `TreeView` raises:
```
[ERROR] .../data/core/init.lua:185: attempt to perform arithmetic on a nil value (local 'directory_start_idx') at .../data/core/init.lua:1224

stack traceback:
.../data/core/init.lua:185: in function <.../data/core/init.lua:181>
(...tail calls...)
.../data/plugins/treeview.lua:438: in method 'toggle_expand'
.../data/plugins/treeview.lua:607: in field 'perform'
.../data/core/command.lua:90: in function <.../data/core/command.lua:83>
[C]: in function 'xpcall'
.../data/core/init.lua:1223: in function 'core.try'
.../data/core/command.lua:102: in function 'core.command.perform'
.../data/plugins/treeview.lua:630: in field 'perform'
.../data/core/command.lua:90: in function <.../data/core/command.lua:83>
[C]: in function 'xpcall'
...(skipping 1 levels)
.../data/core/keymap.lua:205: in function 'core.keymap.on_key_pressed'
.../data/core/keymap.lua:246: in function 'core.keymap.on_mouse_pressed'
.../data/core/init.lua:1248: in upvalue 'on_event'
.../data/plugins/macro.lua:19: in function 'core.on_event'
[C]: in function 'xpcall'
.../data/core/init.lua:1223: in function 'core.try'
.../data/core/init.lua:1308: in function 'core.step'
.../data/core/init.lua:1386: in function 'core.run'
(...tail calls...)
[string "local core..."]:9: in function <[string "local core..."]:2>
[C]: in function 'xpcall'
[string "local core..."]:2: in main chunk
```

Potential fix:
```diff
diff --git a/data/core/init.lua b/data/core/init.lua
index 89b7725..03a04d7 100644
--- a/data/core/init.lua
+++ b/data/core/init.lua
@@ -170,6 +170,8 @@ local function project_subdir_bounds(dir, filename, start_index)
       return not common.path_belongs_to(file.filename, filename)
     end, start_index + 1)
     return start_index, end_index - start_index, dir.files[start_index]
+  else
+    return start_index, 0
   end
 end
```

--- a/data/core/dirwatch.lua
+++ b/data/core/dirwatch.lua
@@ -166,7 +166,7 @@ end
 
 
 local function compare_file(a, b)
-  return a.filename < b.filename
+  return system.path_compare(a.filename, a.type, b.filename, b.type)
 end
 
 

